### ISSUE TITLE
Add support to specify content encoding in upload task

### ISSUE THREAD

**Issue by atrauzzi** [aws/aws-vsts-tools#114]
Support for content type was added here: https://github.com/aws/aws-vsts-tools/issues/3#issuecomment-323180923

Although I'm pre-compressing my javascript and need to be able to serve it with a custom content encoding (`gzip`) header.  Any chance this can be added as another option for any tasks that put files on S3?

**Comment by steveataws** [comment]
Yeah, we can take a look - we'd need to have the setting apply to all files in the upload so if you have multiple different file types to upload, each with their own setting, you'd need to run multiple targeted upload tasks. I assume this would be ok?

**Comment by atrauzzi** [comment]
Definitely, that makes sense.  I already handle that with my shell script equivalents:

```
    aws s3 sync \
        "${AM_TEMP}/nodejs/build/release" \
        "s3://project-${ENVIRONMENT}-assets" \
        --exclude "*" \
        --include "*.js" \
            --acl public-read \
            --storage-class REDUCED_REDUNDANCY \
            --content-encoding gzip

    aws s3 sync \
        "${AM_TEMP}/nodejs/build/release" \
        "s3://project-${ENVIRONMENT}-assets" \
        --include "*" \
        --exclude "*.js" \
            --acl public-read \
            --storage-class REDUCED_REDUNDANCY
```

I can only assume that's what you're going to end up doing behind the scenes. 😄

**Comment by hunterwerlla** [comment]
This is fixed as of the 1.2.0 release
